### ISSUE TITLE
[MNG-4166] Problem parsing command-line options in release:perform

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5590]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** opened **[MNG-4166](https://issues.apache.org/jira/browse/MNG-4166?redirect=false)** and commented

From Paul Merlin on users@maven.apache.org (http://www.nabble.com/Re%3A--PLEASE-TEST--Maven-2.2.0-RC2-p23391250.html)

```
[INFO] [release:perform]                                                                                                                  
[INFO] Change the default 'svn' provider implementation to 'javasvn'.                                                                    
[INFO] Checking out the project to perform the release ...                                                                                
[INFO] SVN checkout directory: /home/paul/tmp/ekPom/target/checkout                
[INFO] Executing goals 'deploy'...                                                                                                        
[INFO]
------------------------------------------------------------------------                                                          
[ERROR] BUILD ERROR                                                                                                                      
[INFO]
------------------------------------------------------------------------                                                          
[INFO] Failed to re-parse additional arguments for Maven invocation.                                                                      

Unrecognized option: -f
[INFO]
------------------------------------------------------------------------
[INFO] Trace                                                                  
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to re-parse
additional arguments for Maven invocation.
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:703)      
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeStandaloneGoal(DefaultLifecycleExecutor.java:553)
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:523)          
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:371)
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:268)        
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:181)                    
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:356)                                                    
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:137)                                                      
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:356)                                                              
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                                        
        at
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)                                      
        at
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)                              
        at java.lang.reflect.Method.invoke(Method.java:597)                                                                  
        at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)                                                
        at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)                                                        
        at
org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)                                              
        at org.codehaus.classworlds.Launcher.main(Launcher.java:375)                                                          
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to re-parse
additional arguments for Maven invocation.      
        at
org.apache.maven.plugins.release.PerformReleaseMojo.execute(PerformReleaseMojo.java:133)                          
        at
org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:483)                            
        at
org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:678)                
        ... 16 more                                                                                                          
Caused by: org.apache.maven.shared.release.ReleaseExecutionException: Failed
to re-parse additional arguments for Maven invocation.
        at
org.apache.maven.shared.release.phase.AbstractRunGoalsPhase.execute(AbstractRunGoalsPhase.java:89)                      
        at
org.apache.maven.shared.release.phase.RunPerformGoalsPhase.execute(RunPerformGoalsPhase.java:67)                        
        at
org.apache.maven.shared.release.DefaultReleaseManager.perform(DefaultReleaseManager.java:336)                          
        at
org.apache.maven.shared.release.DefaultReleaseManager.perform(DefaultReleaseManager.java:282)                          
        at
org.apache.maven.shared.release.DefaultReleaseManager.perform(DefaultReleaseManager.java:262)                          
        at
org.apache.maven.plugins.release.PerformReleaseMojo.execute(PerformReleaseMojo.java:129)                                
        ... 18 more                                                                                                                
Caused by: org.apache.maven.shared.release.exec.MavenExecutorException: Failed
to re-parse additional arguments for Maven invocation.
        at
org.apache.maven.shared.release.exec.InvokerMavenExecutor.setupRequest(InvokerMavenExecutor.java:335)                    
        at
org.apache.maven.shared.release.exec.InvokerMavenExecutor.executeGoals(InvokerMavenExecutor.java:377)                    
        at
org.apache.maven.shared.release.exec.InvokerMavenExecutor.executeGoals(InvokerMavenExecutor.java:413)                    
        at
org.apache.maven.shared.release.phase.AbstractRunGoalsPhase.execute(AbstractRunGoalsPhase.java:81)                        
        ... 23 more                                                                                                                  
Caused by: org.apache.commons.cli.UnrecognizedOptionException: Unrecognized
option: -f                                              
        at org.apache.commons.cli.Parser.processOption(Parser.java:363)                                                              
        at org.apache.commons.cli.Parser.parse(Parser.java:199)                                                                      
        at org.apache.commons.cli.Parser.parse(Parser.java:85)                                                                      
        at
org.apache.maven.shared.release.exec.InvokerMavenExecutor.setupRequest(InvokerMavenExecutor.java:186)                    
        ... 26 more                                                                                                                  
[INFO] 
```

---

**Affects:** 2.2.0

**Issue Links:**
- [MNG-4285](https://issues.apache.org/jira/browse/MNG-4285) CliManager.parse() cannot be called from within a mojo

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

commons-cli version 1.2 is more strict about extraneous cli options. the maven artifact filter manager was forcing plugins to use the core version of commons-cli, so this was messing up the release plugin. We're still using version 1.2 in the core, but I've removed the commons-cli line from the artifact filter manager to allow plugins to use their own versions of the library.

**Comment by jira-importer** [comment]
**[Venkatesh W R](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=venkateshwr)** commented

Hi John

When using maven release plugins with maven 2.1 version, I had encountered similar issues. I see in the maven 2.2 release notes that this issue has been resolved. I downloaded maven 2.2 and tried to execute the maven release goals. I wanted to pass 2 parameters to the "arguments" parameter of release plugin so that it is used within the maven goals. Unfortunately, the error still remains. I am not sure if I am doing something wrong here.
Here is the maven command

/opt/build/maven-2.2/bin/mvn -B -e  release:clean release:prepare release:perform -DpreparationGoals=install -DremoteTagging=false -DautoVersionSubmodules=true -Dtag=1.0.1 -DreleaseVersion=1.0.1 -DjavaHome=/usr/java/jdk1.6.0_10/bin -DmavenHome=/opt/build/maven-2.2/bin -Darguments="-DaltDeploymentRepository=ig.releases::default::http://somehost:8081/nexus/content/repositories/ig.release/ -Dmaven.test.skip=true"

This is the snippet of the error I see

INFO 2009-07-07 07:13:15,798 - OUTPUT>[INFO] Executing goals 'install'...
INFO 2009-07-07 07:13:15,804 - OUTPUT>[INFO] ------------------------------------------------------------------------
INFO 2009-07-07 07:13:15,804 - OUTPUT>[ERROR] BUILD ERROR
INFO 2009-07-07 07:13:15,804 - OUTPUT>[INFO] ------------------------------------------------------------------------
INFO 2009-07-07 07:13:15,809 - OUTPUT>[INFO] Failed to re-parse additional arguments for Maven invocation.
INFO 2009-07-07 07:13:15,809 - OUTPUT>
INFO 2009-07-07 07:13:15,809 - OUTPUT>unbalanced quotes in "-DaltDeploymentRepository=ig.releases::default::http://somehost:8081/nexus/content/repositories/ig.release/ -P env-properties,env-properties
INFO 2009-07-07 07:13:15,809 - OUTPUT>[INFO] ------------------------------------------------------------------------
INFO 2009-07-07 07:13:15,809 - OUTPUT>[INFO] Trace
INFO 2009-07-07 07:13:15,809 - OUTPUT>org.apache.maven.lifecycle.LifecycleExecutionException: Failed to re-parse additional arguments for Maven invocation.
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:703)
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeStandaloneGoal(DefaultLifecycleExecutor.java:553)
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:523)
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:371)
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:268)
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:181)
INFO 2009-07-07 07:13:15,809 - OUTPUT> at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:356)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:137)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.apache.maven.cli.MavenCli.main(MavenCli.java:362)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.apache.maven.cli.compat.CompatibleMain.main(CompatibleMain.java:41)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at java.lang.reflect.Method.invoke(Method.java:585)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
INFO 2009-07-07 07:13:15,810 - OUTPUT> at org.codehaus.classworlds.Launcher.main(Launcher.java:375)
INFO 2009-07-07 07:13:15,810 - OUTPUT>Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to re-parse additional arguments for Maven invocation.
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:483)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:678)
INFO 2009-07-07 07:13:15,811 - OUTPUT> ... 17 more
INFO 2009-07-07 07:13:15,811 - OUTPUT>Caused by: org.apache.maven.shared.release.ReleaseExecutionException: Failed to re-parse additional arguments for Maven invocation.
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.phase.AbstractRunGoalsPhase.execute(AbstractRunGoalsPhase.java:89)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.phase.RunPrepareGoalsPhase.execute(RunPrepareGoalsPhase.java:42)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.DefaultReleaseManager.prepare(DefaultReleaseManager.java:196)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.DefaultReleaseManager.prepare(DefaultReleaseManager.java:133)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.DefaultReleaseManager.prepare(DefaultReleaseManager.java:96)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.plugins.release.PrepareReleaseMojo.execute(PrepareReleaseMojo.java:182)
INFO 2009-07-07 07:13:15,811 - OUTPUT> ... 19 more
INFO 2009-07-07 07:13:15,811 - OUTPUT>Caused by: org.apache.maven.shared.release.exec.MavenExecutorException: Failed to re-parse additional arguments for Maven invocation.
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.exec.InvokerMavenExecutor.setupRequest(InvokerMavenExecutor.java:335)
INFO 2009-07-07 07:13:15,811 - OUTPUT> at org.apache.maven.shared.release.exec.InvokerMavenExecutor.executeGoals(InvokerMavenExecutor.java:377)
INFO 2009-07-07 07:13:15,812 - OUTPUT> at org.apache.maven.shared.release.exec.InvokerMavenExecutor.executeGoals(InvokerMavenExecutor.java:413)
INFO 2009-07-07 07:13:15,812 - OUTPUT> at org.apache.maven.shared.release.phase.AbstractRunGoalsPhase.execute(AbstractRunGoalsPhase.java:81)
INFO 2009-07-07 07:13:15,812 - OUTPUT> ... 24 more
INFO 2009-07-07 07:13:15,812 - OUTPUT>Caused by: org.codehaus.plexus.util.cli.CommandLineException: unbalanced quotes in "-DaltDeploymentRepository=ig.releases::default::http://somehost:8081/nexus/content/repositories/ig.release/ -P env-properties,env-properties
INFO 2009-07-07 07:13:15,812 - OUTPUT> at org.codehaus.plexus.util.cli.CommandLineUtils.translateCommandline(CommandLineUtils.java:410)
INFO 2009-07-07 07:13:15,812 - OUTPUT> at org.apache.maven.shared.release.exec.InvokerMavenExecutor.setupRequest(InvokerMavenExecutor.java:185)
INFO 2009-07-07 07:13:15,812 - OUTPUT> ... 27 more
INFO 2009-07-07 07:13:15,812 - OUTPUT>[INFO] ------------------------------------------------------------------------
INFO 2009-07-07 07:13:15,812 - OUTPUT>[INFO] Total time: 4 seconds
INFO 2009-07-07 07:13:15,812 - OUTPUT>[INFO] Finished at: Tue Jul 07 07:13:15 BST 2009
INFO 2009-07-07 07:13:16,013 - OUTPUT>[INFO] Final Memory: 24M/323M
INFO 2009-07-07 07:13:16,013 - OUTPUT>[INFO] ---------------------------------------------------------------

Any help would be much appreciated.

Thanks
Venkatesh
